### PR TITLE
Feature/particle system curves

### PR DESCRIPTION
**Pull request by yalania**
- Gravity for particles,
- Velocity changes for particles
- Super boost on particle system rendering time

### CODE REVIEW

File: Engine/Component/ComponentParticleSystem.cpp [hunk 1]
```diff
@@ -24,6 +25,8 @@ ComponentParticleSystem::~ComponentParticleSystem()
 
 void ComponentParticleSystem::Init() 
 {
+	glGenBuffers(1, &ssbo);
+	shader_program = App->program->GetShaderProgramId("Particles");
```
Comment (raulgonzalezupc) [inline Engine/Component/ComponentParticleSystem.cpp:29]: if you are going for the approach of creating one shader for PS, then you should send the uniforms on the componentParticleSysyem, not on billboard

File: Engine/Component/ComponentParticleSystem.cpp [hunk 2]
```diff
@@ -163,53 +193,83 @@ void ComponentParticleSystem::Render()
...
-				{
-					billboard->Render(p.position);
-				}
-				
+				std::iter_swap(particles.begin() +i, particles.begin()+ num_of_alive_particles);
+				++num_of_alive_particles;
 			}
+
 		}
+		billboard->CommonUniforms(shader_program);
```
Comment (raulgonzalezupc) [inline Engine/Component/ComponentParticleSystem.cpp:223]: the uniforms should be on this cpp

File: Engine/Component/ComponentParticleSystem.cpp (hunk 1 above)
Comment (yalania) [reply Engine/Component/ComponentParticleSystem.cpp:29]: but this way I will be duplicating code.

File: Engine/Component/ComponentParticleSystem.cpp (hunk 2 above)
Comment (mariofv) [reply Engine/Component/ComponentParticleSystem.cpp:223]: Dont worry about this too much, as this changes with uber

**Review by mariofv — APPROVED** [review]
All good!

**Review by raulgonzalezupc — APPROVED** [review]

**Pull request merged**